webapp/tests/forms/steps/test_lotse_steuerlotse_steps.py

```diff
@@ -0,0 +1,94 @@
+import pytest
+from werkzeug.datastructures import MultiDict
+
+from app.elster_client.elster_client import request_tax_offices
+from app.forms.steps.lotse.lotse_steuerlotse_steps import StepSteuernummer, LotseFormSteuerlotseStep
+
+
+class SummaryStep(object):
+    pass
+
+
+class TestLotseFormSteuerlotseStep:
+    @pytest.fixture(autouse=True)
+    def attach_fixtures(self, app, test_request_context):
+        self.app = app
+        self.req = test_request_context
```

Comment:
The `attach_fixtures` pattern is only necessary for adding pytest fixtures to unittest class-based test suites. I wouldn't recommend it for pure pytest tests.

Instead, specify on the test function itself which fixtures are used.

Three variants:

1) If you need the value of the fixture in the test, ["request" it by passing it as a parameter](https://docs.pytest.org/en/6.2.x/fixture.html#requesting-fixtures):
```py
def test_whatever(app):
    assert app.something() == 1
```

2) If you only need the side-effect **in a single test**, [use a decorator on the test](https://docs.pytest.org/en/6.2.x/fixture.html#use-fixtures-in-classes-and-modules-with-usefixtures):
```py
@pytest.mark.usefixtures("app")
def test_whatever():
    assert 2+2 = 5
```

3) If you only need the side-effect **for all tests a class**, decorate the class. Note that you can still request the fixture in individual tests if you need the value there.
```py
@pytest.mark.usefixtures("app")
class TestStuff:
    def test_whatever(self):
        assert 2+2 = 5
    def test_using_value(self, app):
        assert app.something() == 1
```

Reply:
That makes so much more sense. Thank you for explaining that